### DISCUSSION TITLE
Setup the own server DB.

### DISCUSSION THREAD

**Discussion by SWS-5007** [polkascan/explorer#20] [Q&A] [answered]
Hi, I am going to  setup the explorer in my local.
But I am going to use the DB of the Server.
Here is Server Database Infos.
```
Server Url: https://81.181.255.414
DB Name: test_explorer
DB PORT=3306
```

How should I setup the `DB_CONNECTION` to connect to server DB?
Current `DB_CONNECTION` environment is like this:
`DB_CONNECTION=mysql+pymysql://root:root@mysql:3306/polkascan?charset=utf8mb4`
How should I change this?

Please help me!

**Comment by arjanz** [comment]
With the info that you provided, that would be:

`DB_CONNECTION=mysql+pymysql://username:password@81.181.255.414:3306/test_explorer?charset=utf8mb4`

Mind you that this is just for the `harvester` service in the Docker compose file. You will have to make the same changes in the `API_SQLA_URI` env variable in the `api` service. 

Also you will need two databases, one for the raw harvester data and one for the explorer application where the ETL processes will put the specific explorer data (in the default Docker compose set to `polkascan_explorer`.)

**Reply by SWS-5007** [reply]
Hi, Thanks for your helps.
But I have some questions yet.

So Should I create two database?
For example, `test_explorer` & `test_db`?

**Reply by SWS-5007** [reply]
And I am trying this guide, [https://github.com/polkascan/harvester/tree/main#running-locally](https://github.com/polkascan/harvester/tree/main#running-locally) to run polkascan on my local.
But when I run this command,`alembic upgrade head` I am getting error like below.

>   File "C:\Users\Home\AppData\Local\Programs\Python\Python310\lib\site-packages\pymysql\connections.py", line 748, in _read_bytes
>     raise err.OperationalError(
> sqlalchemy.exc.OperationalError: (pymysql.err.OperationalError) (2013, 'Lost connection to MySQL server during query')
> (Background on this error at: https://sqlalche.me/e/14/e3q8)

What is wrong?
How can I fix this problem?

**Reply by arjanz** [reply]
In your example the DB_CONNECTION for the `harvester` service should point to `test_db`:
`DB_CONNECTION=mysql+pymysql://test_root:6kkgyRfrdLy=@81.181.255.414:3306/test_db?charset=utf8mb4`

**Reply by arjanz** [reply]
`'Lost connection to MySQL server during query'` seems like connectivity problems?

**Comment by SWS-5007** [comment, marked as answer]
Hi, I am getting  `(2013, 'Lost connection to MySQL server during query')` error yet when I run the `alembic upgrade head` command.

This means, It can't connect to MySQL Database which I gave for `DB_CONNECTION` env with `db_name`, `db_pwd` and `db_url`?

Here is my codebase of local_settings.py of havester.

```
DB_CONNECTION = "mysql+pymysql://shearscan_root:6kkgyRfrdLy=@81.181.255.414:3306/shearscan_db?charset=utf8mb4"
SUBSTRATE_RPC_URL = "wss://www.shear.com/"

INSTALLED_ETL_DATABASES = []
```

And I created the Database called `shearscan_db` on Cpanel.
![image](https://user-images.githubusercontent.com/119567356/213004673-384acb73-b7fb-4c3b-b7c4-10e8d0cf9978.png)

And also db_name and db_pwd is fitted.
My Cpanel url is [81.181.255.414](https://81.181.255.414).

What is wrong in my env configuration?
Please.

**Reply by SWS-5007** [reply]
I have fixed this problem.
I asked Cpanel Support Team to allow 3306 port for me.
Then, It works well.
Thanks so much.
